### ISSUE TITLE
Container for your app

### ISSUE THREAD

**Issue by Manuel-DominguezCBG** [broadinstitute/SpliceAI-lookup#51]
I have finally make the app work in a docker container. If you needed please use it
https://hub.docker.com/r/monkiky/spliceai-lookup

Do you have available the back-end so that I can replicate locally as seen online?

**Comment by Manuel-DominguezCBG** [comment]
I have finally make the app work in a docker container. If you needed please use it
https://hub.docker.com/r/monkiky/spliceai-lookup

Do you have available the back-end so that I can replicate locally as seen online?

**Comment by bw2** [comment]
I'm not sure I understand your question. All code for the backend is available in this github repo.

**Comment by Manuel-DominguezCBG** [comment]
Sorry for this misunderstanding. I explain again.

In this repository, I am able to run the app locally and in the web browser get a response like this 

![image](https://github.com/broadinstitute/SpliceAI-lookup/assets/74112779/1c627255-60c4-4c1c-8d7a-fb6f04cd030b)

Now, what I would like to do is to replicate the same output obtained when I run your app online. This:

![image](https://github.com/broadinstitute/SpliceAI-lookup/assets/74112779/c5677f70-9e98-44db-919d-232f7ff0a708)

Where is this? If this is in this repository I cannot find it or make it work.

Thanks one more time for your continued help.

(Sorry in advance for not using the right terminology, this has been the first time I have worked with API, dockers, app and so on)

**Comment by bw2** [comment]
@Manuel-DominguezCBG I just made some updates to the code that might make this easier (along with other changes). 

If you can update your local version, rerun 
`python3 -m pip install -r requirements.txt` to install my latest fork for SpliceAI, 
and then run this script after editing the parameters in there as needed:
`./start_local_server.sh`

The entire website (ie. front-end) is contained in the index.html file. 

To view it, you should have the ./start_local_server.sh script running in the background, and then go to http://localhost:8080 in your Chrome browser (assuming Chrome is open on the same computer that's running the server script, or if not, then go to http://<ip address of server machine>/index.html). You should then be able to enter a variant and, when you hit submit, it will send a request to your server process to compute the score. If this doesn't work, and the page isn't sending the request to the right address, you can manually change the address by editing your local version of index.html on this line:
https://github.com/broadinstitute/SpliceAI-lookup/blob/master/index.html#L9

In Chrome, you can also open the Javascript Console before loading the page using this menu:
<img width="592" alt="image" src="https://github.com/broadinstitute/SpliceAI-lookup/assets/6240170/e5672e0b-58dc-4be7-8144-b39524bddb4d">

That way you can see log messages and make sure the website is loading correctly. With the Javascript Console, it should look like:

<img width="1792" alt="image" src="https://github.com/broadinstitute/SpliceAI-lookup/assets/6240170/0bfda21e-f54d-403c-8498-830692de9e32">


There are additional somewhat-complicated steps if you want the website to use a secure url (eg. https:// instead of http://).

**Comment by Manuel-DominguezCBG** [comment]
Thank you soo much!!! I will try this today and I will come back to you. Please don't close the issue yet.

**Comment by Manuel-DominguezCBG** [comment]
Everything working BUT I got this 
![Screenshot 2023-09-05 at 10 43 34](https://github.com/broadinstitute/SpliceAI-lookup/assets/74112779/45d7012c-2490-4502-9fb6-da8e628947e6)

 I am not sure where the problem is

**Comment by bw2** [comment]
Can you try running 
python3 -m pip uninstall spliceai
python3 -m pip install -r requirements.txt

**Comment by Manuel-DominguezCBG** [comment]
Yes I did that. I have installing right now your repository in a new computer from scratch and I got the same warning. I have noticed a warning

**Comment by bw2** [comment]
If you're still seeing the same issue on the website, can you share a screenshot of the log messages in the Chrome Javascript Console after the page loads?

**Comment by Manuel-DominguezCBG** [comment]
Now is working in a new computer. what I have done is to no use python 3.6 as you use. I will record all changes and debugging problems. It was a madness regarding the tensorflow, keras, settools and pyvcf incompatibility. Please dont need to do anything so far. I need to rebuild this again (becaouse the previous container was build with the python 3.6 version). 

I will update you and hopefully I can have a fully operative container working for my team.

**Comment by bw2** [comment]
Glad it's working! 

On my MacOS laptop, I also have it working with python3.9 with 
keras==2.13.1
tensorflow==2.13.0

I also had problems with the pyvcf install, and had to separately downgrade to setuptools==57.5.0, install pyvcf, then upgrade setuptools.

**Comment by Manuel-DominguezCBG** [comment]
I am building the container now (finger cross).

Thank you for your help. When working I will put the container in the Docker hub and I will give you the link. I will also put all the details in the description file of the container. I have also made some improvements regarding installing Python packages, now Torch is in the requirements.txt file to make sure all can be installed in a single command.

I will close the issue when I have a fully operative container ready to share.

**Comment by Manuel-DominguezCBG** [comment]
Hi, 

The local container I have created with your local Spliceai lookup works perfectly in our clinical lab. I just wanted to say thank you for all the help received.

I am going to upload the container in the docker hub. If you want the code with the modifications  I did in your code to make the container work let me know and I can fork your repository and add these changes. I can also give you the link to the docker container.

regards, 

manuel

**Comment by bw2** [comment]
Hi Manuel, 

That's great that you're done with the container! It was good timing because we are now working on a major update (as you may have seen in recent commits). Among other things, it will change how transcript annotations are updated and stored on the server. I would recommend not updating your version yet, until we're done with this round of updates.

One thing I'm curious about - what were the main reasons your site decided to set up a local instance of SpliceAI-lookup?

-Ben

**Comment by Manuel-DominguezCBG** [comment]
Hi Ben,

In our lab, we do clinical variant interpretation and we are very serious regarding reproducibility of our analysis. We have Standard Operarating Procedures for the analysis we carry out and bioinformaticians care version control. I noticed that your work already does what we need. If my team use your online version your future modification will requiere me to check if the new modification satisfy our need and doesn't impact patient care. For instance, the new modification you mentioned in your last message likely will requiered me to validate the tool again and it may need to change the SOP clinical scientists follow and this is a lot of work.

We also like having control of the tool we use.For intance after having the local program working, I am going to introduce modifications to satisfy the specific need of my teams such as modifying the transcripts so that annotation is giving by specific transcripts instead the large list of transcripts your program normally provides. 

I will be vigilant to your progress to see if new updates can be usefull to me team. You are doing a great job. I was Monday and yesterday in a national UK conference of clinical genomics and I saw  a lot of screenshots people took from your lookup online version to show results. In my lab people love it too.

Finally, I am a nurse and a biologist going to bioinformatics and I have recently found programing one of the most exciting professional things in my life. Is there is a small and not urgent work you need in this project I am happy to consider it to do for free in my free time.

**Comment by bw2** [comment]
That's really interesting. Thanks for the detailed response and the offer to help. There might be some projects like that after we finish the current round of large-scale updates.

**Comment by Manuel-DominguezCBG** [comment]
Thanks Ben, 

Please let me know if you think I can help. 

I close the issue.

Regards

Manuel

**Comment by bw2** [comment]
In case you'd like to take a look at the dev. version with upcoming features and visualizations, it's @

https://broadinstitute.github.io/SpliceAI-lookup-dev/index.html#variant=2%3A178581491%20ATACT%3EA&hg=38&distance=500&mask=1

Let me know if you find any issues there.

Thanks,
-Ben

**Comment by Manuel-DominguezCBG** [comment]
Thanks for that I will save the link to check what is new.

At the moment, the web is not connecting with the API. I get this message 

SpliceAI API call failed: Failed to fetch

Pangolin API call failed: Failed to fetch
